src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/HttpResponseMessageTestExtensions.cs

```diff
@@ -14,15 +15,24 @@ public static void AssertOk(this HttpResponseMessage message)
         Assert.Equal(OK, message.StatusCode);
     }
 
-    public static T AssertOk<T>(this HttpResponseMessage message)
+    public static T AssertOk<T>(this HttpResponseMessage message, bool useSystemJson = false)
     {
         Assert.Equal(OK, message.StatusCode);
 
-        var body = message.Content.ReadFromJson<T>();
+        T? body = Deserialize<T>(message, useSystemJson);
 
         return Assert.IsType<T>(body);
     }
 
+    private static T? Deserialize<T>(this HttpResponseMessage message, bool useSystemJson)
```

Comment:
We'll need to use this in `AssertBodyEqualTo<T>` as well i.e.

```cs
public static T AssertBodyEqualTo<T>(this HttpResponseMessage message, T expected, bool useSystemTextJson = false)
{
    var body = Deserialize<T>(message, useSystemJson: useSystemTextJson);

    // ...
}
```

This also means we need changes in `AssertOk<T>` and `AssertCreated<T>` to provide `useSystemJson` flags with defaults

Reply:
done :)